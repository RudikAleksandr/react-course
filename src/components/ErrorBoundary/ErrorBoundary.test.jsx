import React from 'react';
import { render } from '@testing-library/react';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary component', () => {
  test('render ErrorBoundary snapshot', () => {
    const { asFragment } = render(<ErrorBoundary>App</ErrorBoundary>);

    expect(asFragment(<ErrorBoundary />)).toMatchSnapshot();
  });

  test('should render ErrorBoundary when there is an error', () => {
    const Child = () => {
      throw new Error();
    };

    const { getByText } = render(<ErrorBoundary><Child /></ErrorBoundary>);
    const errorMessage = getByText('Something went wrong!');
    expect(errorMessage).toBeDefined();
  });
});
