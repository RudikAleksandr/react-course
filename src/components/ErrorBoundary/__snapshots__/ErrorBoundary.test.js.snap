// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component render ErrorBoundary snapshot 1`] = `
<DocumentFragment>
  App
</DocumentFragment>
`;
